### ISSUE TITLE
[a11y]: When an error occurs on clicking CreateTearsheetStep Next button, keyboard focus moves away from CreateTearsheetStep

### ISSUE THREAD

**Issue by mariyageorge01** [carbon-design-system/ibm-products#7691]
### Package

@carbon/ibm-products

### Browser

Chrome

### Operating System

MacOS

### Package version

2.64.0

### React version

18.3.1

### Automated testing tool and ruleset

Keyboard navigation

### Description

If an error occurs on clicking the Next button in CreateTearsheetStep, the keyboard focus moves away from the CreateTearsheetStep.

### WCAG 2.1 Violation

2.1.1 Keyboard

### Reproduction/example

https://stackblitz.com/github/carbon-design-system/ibm-products/tree/main/examples/carbon-for-ibm-products/CreateTearsheet?file=src%2FExample%2FExample.jsx

### Steps to reproduce

1. In the above stackblitz reproduction, the simulate error toggle should be on. 
2. Click on the Next button and you can see that the focus has moved away from the CreateTearsheetStep.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate problems

**Comment by amal-k-joy** [comment]
@mariyageorge01 ,
It looks like you're not passing the `hasError` prop to CreateTearsheet, which would enable the [default behavior](https://stackblitz.com/edit/github-pplcxgzt?file=src%2FExample%2FExample.jsx) of automatically focusing the first visible element when the component opens. 

If you prefer to override this behavior, you can do so using a setTimeout like this:
This allows you to manually control which element receives focus after an error occurs.


```
if (shouldReject) {
  setHasSubmitError(true);
  setTimeout(() => {
    document
      .querySelector('.c4p--tearsheet-create__create-button')
      ?.focus();
  }, 1);
  reject('Simulated error');
}
```
